### PR TITLE
update: optimize LRUCache memory usage (#1815)

### PR DESCRIPTION
**Pull request by fansehep**
Signed-off-by: fan <yfan3763@gmail.com>

### What problem does this PR solve?

Issue Number: #xxx 

Problem Summary:
  Old Design will make the Key construct function call twice.
### What is changed and how it works?

What's Changed:
 I change the Item key to be a key pointer, std::unordered_map will save the key, so just make the Item.key point to the std::unordered_map<>::iterator->key. 
  By the way, **This LRUCache is not standard**. It's bad to change a standard datastruct for business needs. We can custom a  datastruct for business needs. Maybe some other file just need esay interface such as get or put. It don't need like this "LRUCache" provide some function like GetLast().
How it Works:

Side effects(Breaking backward compatibility? Performance regression?):

### Check List

- [ ] Relevant documentation/comments is changed or added
- [ ] I acknowledge that all my contributions will be made under the project's license

### CODE REVIEW

File: src/common/lru_cache.h [hunk 1]
```diff
@@ -139,8 +139,10 @@ template <typename K,  typename V,
     typename ValueTraits = CacheTraits<V>>
 class LRUCache : public LRUCacheInterface<K, V> {
  public:
+    // to less one Key construct
+    // make the Item key be a pointer
     struct Item {
-        K key;
+        K* key;
```
Comment (wu-hanqing) [inline src/common/lru_cache.h:145]: `const K* key` ?

Comment (fansehep) [reply src/common/lru_cache.h:145]: fix

**Comment by ilixiaocui** [comment]
Please merge historical commits into one.

**Comment by fansehep** [comment]
> Please merge historical commits into one.

so but there is only one? right?

**Comment by fansehep** [comment]
> > Please merge historical commits into one.
> 
> so but there is only one? right?

**Comment by fansehep** [comment]
recheck

**Pull request closed without merging**